tinygrad/runtime/graph/hcq.py

```diff
@@ -52,6 +52,7 @@ def __init__(self, jit_cache: List[ExecItem], input_rawbuffers: List[Buffer], va
 
     for j,ji in enumerate(jit_cache):
       enqueue_dev = ji.prg.dev if (is_exec_prg:=isinstance(ji.prg, CompiledRunner)) else Device[ji.bufs[1].device] #type:ignore
+      assert isinstance(enqueue_dev, HCQCompiled) and (isinstance(enqueue_dev.hw_copy_queue_t, type) and enqueue_dev.hw_copy_queue_t is HWQueue)
```

Comment:
The second expression of this assert could also be replaced with `enqueue_dev.hw_copy_queue_t is not None`, which is however less narrow.